src/Chainweb/Payload.hs

```diff
@@ -516,6 +556,24 @@ instance IsCasValue TransactionTree where
     type CasKeyType TransactionTree = BlockTransactionsHash
     casKey = _transactionTreeHash
 
+instance ToJSON TransactionTree where
+    toJSON o = object
+        [ "hash" .= _transactionTreeHash o
+        , "tree" .= merkleTreeToJson (_transactionTree o)
```

Comment:
I cleaner/more idiomatic solution may be to use a newtype wrapper with the respective JSON instance. I think, I just wanted to avoid the overhead of introducing a `newtype`. The actuall code wouldn't change that much.

Reply:
ok